package/MDAnalysis/analysis/MSD.py

```diff
@@ -0,0 +1,468 @@
+(MSD) as a function of time, averaging over all atoms in the atom selection
+and all restarts. The MSD of any atom selection can be calculated,
+as any atoms which remain within the atom group over an entire time interval
+are included in the calculation, allowing for highly specific, position-
+dependent MSD calculations. The tool is robust to zero-length atom selections,
+and allows for the calculation of multiple atom selections simultaneously.
+Additional command line functionality is built in to allow for easy use
+in a batch/slurm scheduler and parallelization of especially large jobs.
+
+Important Note: Calculations are run on the trajectory as given to the tool,
+so some preprocessing may be required to properly account for periodic
+boundary conditions. For example, if calculating the MSD of a Gromacs
+simulation, the following preprocessing is necessary:
+    >>> gmx trjconv -f mysim.xtc -s mysim.tpr -o processed_sim.xtc -pbc nojump
+
+MSD command line usage
+----------------------
+
+This calculates the MSD of 3 atom selections over the first 50000 frames
+of the trajectory:
+
+    >>> ./MSD.py -f mysim.xtc -s mysim.tpr --sel 'name NA,name NA and prop z
+            > 50 and prop z < 150, name CL' -b 0 -e 50000 --dt 5 -o mysim.p
+            --len 1500
+
+Output will be written to mysim.p in the format [msd,n_samples], where msd
+is a vector of length=3 containing vectors of MSD values with index
+corresponding to time interval, and n_samples is a similar array containing
+the number of samples associated with each index in msd.
+
+The option --len alters the number of calculations performed by the tool. If
+you are taking a linear regression of the MSD vs tau plot over the first 'n'
+frames, adjusting this parameter allows you to collect the data you need,
+allowing you to avoid calculating MSD values at tau >= n+1.
+
+The option --max_data will use data after the frame specified by -e if it is
+available. This allows you to avoid losing data if you need to split the
+calculation over several parallel jobs.
+
+See the main method of MSD.py for an example of python script implementation.
+
+Options:
+--------
+-f name of trajectory file
+-s name of tpr file
+--sel string containing selections separated by commas
+-b t0 (frames)
+-e tf (frames)
+--dt number of frames between restarts
+-o output file name (optional)
+--len maximum number of tau frames to calculate (optional, default=250)
+--out write output boolean (optional, default=True)
+--max_data include data from after tf if it exists
+  this allows for command-line level parallelization without
+  data loss. (optional, default=False)
+"""
+
+
+from __future__ import absolute_import, print_function, division
+
+import sys
+import os
+import getopt
+import warnings
+from collections import deque
+
+import numpy as np
+import pickle
+# from six import range
+
+import MDAnalysis
+from MDAnalysis.lib.distances import squared_distance_vector
+
+
+class MSD(object):
```

Comment:
This script can also be modified to use AnalysisBase but that change isn't straight forward. You can keep it as is right now but we should try to make this look like it was derived from AnalysisBase.